### PR TITLE
Add PyPI integration

### PR DESCRIPTION
**Pull request by rossmacarthur**
image

### CODE REVIEW

**Comment by SplittyDev** [comment]
Great work, thanks a lot for the contribution!
Deploy is pending, will look into it and merge asap 👀

File: README.md [hunk 1]
```diff
@@ -43,6 +44,12 @@ Building SpaceBadgers has been a labor of love, aiming to offer a superior, reli
 
 ## Development
 
+Clone the repository and make sure to initialize the submodules
+
+```bash
+git submodule update --init --recursive
+```
+
```
Comment (SplittyDev) [inline README.md:52]: Thanks for noticing this and including it in the README!

**Comment by SplittyDev** [comment]
Apparently the PyPI response is huge (2MB+):
CleanShot 2023-06-10 at 22 19 32@2x

I just had a look at the docs and it doesn't seem like there's much we can do about that, so I guess we'll have to live with PyPI badges taking slightly longer to generate, since they can't be cached on the edge.

There is an RSS endpoint for getting _just_ the releases, which we could theoretically use to fetch only the latest release:
`https://pypi.org/rss/project/:packageName/releases.xml` ([numpy example](https://pypi.org/rss/project/numpy/releases.xml))

But this also includes `rc` releases and the like, and I'm not sure how to properly differentiate stable releases from release candidates, alpha versions and other variations that probably shouldn't be fetched as `latest`..

I'm fine with merging this as-is, we can always come back later. If you wanna hold off on merging for a bit and look into getting the releases and fetching the latest version only, just let me know.

**Comment by SplittyDev** [comment]
Off-Topic: I just noticed that you're the creator of sheldon! I've been using that tool for quite some time now, thanks a lot for your valuable contributions to the open source community! ✨

**Review by SplittyDev — APPROVED** [review]

**Comment by rossmacarthur** [comment]
> Apparently the PyPI response is huge (2MB+):

Yeah I did notice this locally as well :/. Comparing with a bunch of other packages I'm seeing a big variation in size e.g. `requests` is much much smaller.

```
❯ curl -fsSL -i https://pypi.org/pypi/numpy/json | grep content-length
content-length: 1656988
❯ curl -fsSL -i https://pypi.org/pypi/requests/json | grep content-length
content-length: 163731
❯ curl -fsSL -i https://pypi.org/pypi/scipy/json | grep content-length
content-length: 999371
❯ curl -fsSL -i https://pypi.org/pypi/matplotlib/json | grep content-length
content-length: 1193906
❯ curl -fsSL -i https://pypi.org/pypi/pydantic/json | grep content-length
content-length: 1050036
❯ curl -fsSL -i https://pypi.org/pypi/fastapi/json | grep content-length
content-length: 261320
```

> But this also includes rc releases and the like, and I'm not sure how to properly differentiate stable releases from release candidates, alpha versions and other variations that probably shouldn't be fetched as latest..

I guess we could do this by parsing the version but I'm not sure if Python versions conform to the exact semver format that a Javascript parser can do.

> Please let me know if you want me to merge this as-is (which is totally fine with me, we can always come back to it later), or if you wanna hold off on merging for a bit and look into getting the releases and fetching the latest version only.

I'm happy to merge for now since I think most people will not encounter this unless they have many, many releases for their project. Happy to come back to it later.

**Comment by rossmacarthur** [comment]
> Off-Topic: I just noticed that you're the creator of sheldon! I've been using that tool for quite some time now, thanks a lot for your valuable contributions to the open source community! ✨

Glad you found it useful! Thank you for creating spacebadgers, I've been frustrated with https://shields.io for a while now

File: badgers-web/src/app/page.tsx [hunk 2]
```diff
@@ -180,6 +180,14 @@ export default function Home() {
                     <Row name="Types" path="/npm/types/:pkg" inject={['react']} />
                     <Row name="Types (scoped)" path="/npm/types/:org/:pkg" inject={['@octokit', 'rest']} />
                 </div>
+              </Section>
+              <Section name="PyPI">
+                <div className="grid grid-cols-[auto_1fr_auto] items-center gap-x-8">
+                    <Row name="Name" path="/pypi/name/:pkg" inject={['numpy']} />
+                    <Row name="Version" path="/pypi/version/:pkg" inject={['numpy']} />
+                    <Row name="Name and Version" path="/pypi/info/:pkg" inject={['numpy']} />
+                    <Row name="License" path="/pypi/license/:pkg" inject={['numpy']} />
```
Comment (SplittyDev) [inline badgers-web/src/app/page.tsx:189]: We could use a "lighter" package here to speed up loading of the home page.

```suggestion
                    <Row name="Name" path="/pypi/name/:pkg" inject={['docopt-ng']} />
                    <Row name="Version" path="/pypi/version/:pkg" inject={['docopt-ng']} />
                    <Row name="Name and Version" path="/pypi/info/:pkg" inject={['docopt-ng']} />
                    <Row name="License" path="/pypi/license/:pkg" inject={['docopt-ng']} />
```

**Review by SplittyDev — APPROVED** [review]

**Comment by SplittyDev** [comment]
Maybe we can just replace `numpy` with a package with fewer releases on the homepage? The homepage is server-side rendered, and if the 2MB responses can't be cached that might significantly add to the loading time for every visitor.

A few examples that we could use instead of `numpy`:
- `docopt-ng`
- `pickleDB`
- `janus`

I've gone ahead and added the change as a comment, so you can one-click accept it and avoid having to go back to the code editor. If you wanna use something else, let me know! I'm not super knowledgeable about the Python community, maybe there's more iconic package with fewer releases we could use.

**Comment by rossmacarthur** [comment]
> I've gone ahead and added the suggested change as a review comment. If you wanna use something else, let me know! I'm not super knowledgeable about the Python community, maybe there's a more iconic package with a smaller size.
 
 Interestingly I've noticed that some of the packages I wanted to use don't set the "license" field. E.g. `fastapi`, `seaborn`, but set it in the classifiers. Works on shields.io so I assume they're extracting it from the classifiers.

**Comment by SplittyDev** [comment]
> Interestingly I've noticed that some of the packages I wanted to use don't set the "license" field. E.g. `fastapi`, `seaborn`, but set it in the classifiers. Works on shields.io so I assume they're extracting it from the classifiers.

Ah well, those are a bit annoying to parse. Let's go ahead with merging once CI is green, and I'll do a follow-up for parsing the classifiers to make sure licenses are always working.

**Comment by SplittyDev** [comment]
@all-contributors please add @rossmacarthur for adding PyPI support

**Comment by SplittyDev** [comment]
@all-contributors please add @rossmacarthur for code

**Pull request merged**